packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts

```diff
@@ -0,0 +1,125 @@
+   * @param winnerShares the percent of each respective winning places prize(ie 5 = 5%)
+   * @param stakeId the id of the community stake token, defaults to 2
+   * @param prizeShare the % amount of the prize pool to payout each interval(ie 5 = 5%)
+   * @param voterShare the % amount of prize pool claimable by voters each week(ie 5 = 5%)
+   * @param feeShare the % amount of community stake fees to divert to contest(ie 5 = 5%)
+   * @param weight the weight of each stake balance, defaults to 1
+   * @param walletAddress the wallet address to make transactions from
+   * @returns the contract address of the new contest
+   */
+  async newRecurringContest(
+    namespaceName: string,
+    contestInterval: number,
+    winnerShares: number[],
+    stakeId: number = 2,
+    prizeShare: number,
+    voterShare: number,
+    feeShare: number,
+    weight: number = 1,
+    walletAddress: string,
+  ): Promise<string> {
+    if (!this.initialized || !this.walletEnabled) {
+      await this.initialize(true);
+    }
+    try {
+      const txReceipt = await this.namespaceFactory.newContest(
+        namespaceName,
+        contestInterval,
+        winnerShares,
+        stakeId,
+        prizeShare,
+        voterShare,
+        feeShare,
+        weight,
+        walletAddress,
+      );
+      const newContestAddress =
+        txReceipt.events.NewContest.returnValues.contest;
+      this.contractAddress = newContestAddress;
+      return newContestAddress;
+    } catch (error) {
+      throw new Error('Failed to initialize contest ' + error);
+    }
+  }
+
+  //TODO Reccurring contest pending ABI
+
+  /**
+   * Allows for deposit of contest token(ETH or ERC20) to contest
+   * @param amount amount in ether to send to contest
+   * @param walletAddress the users wallet address
+   * @returns transaction receipt
+   */
+  async deposit(amount: number, walletAddress: string): Promise<any> {
+    this.reInitContract();
+    const tokenAddress = await this.contract.methods.contestToken().call();
+
+    let txReceipt;
+    const weiAmount = this.web3.utils.toWei(amount, 'ether');
+    if (tokenAddress === '0x0000000000000000000000000000000000000000') {
+      //ETH funding route
+      try {
+        txReceipt = await this.contract.methods.deposit(weiAmount).send({
+          value: weiAmount,
+          from: walletAddress,
+          maxPriorityFeePerGas: null,
+          maxFeePerGas: null,
+        });
+      } catch {
+        throw new AppError('ETH transfer failed');
+      }
+    } else {
+      const encodedParameters = this.web3.eth.abi.encodeParameters(
+        ['address', 'uint256'],
+        [this.contractAddress, amount],
+      );
+      const data = `095ea7b3${encodedParameters.substring(2)}`;
+
+      // Create the transaction object
+      const txObject = {
+        to: tokenAddress,
+        data: data,
+        from: walletAddress,
+      };
+      const approveReceipt = this.web3.eth.sendTransaction(txObject);
```

Comment:
'approveReceipt' is assigned a value but never used

Reply:
See the note in the PR. About this helper call. I think we will need to discuss the 2 step transaction flow of this in general.

Reply:
is there anything I can do here or we need some input from design?